src/main/java/it/reply/orchestrator/service/deployment/providers/QcgServiceImpl.java

```diff
@@ -312,10 +314,19 @@ public void cleanFailedUpdate(DeploymentMessage deploymentMessage) {
 
   @Override
   public boolean isUndeployed(DeploymentMessage deploymentMessage) throws DeploymentException {
-    // Nothing to wait here... All the jobs are delete immediately.
+    // Nothing to wait here... All the jobs are deleted immediately.
     return true;
   }
 
+  @Override
+  public void doProviderTimeout(DeploymentMessage deploymentMessage) {
+    if (!deploymentMessage.isPollComplete()) {
```

Comment:
If we get here method must throw, so no `if` is needed

Reply:
You must try. In the previous configuration, when the timer was not on the boundary, control was necessary because the trigger, once fired, remained in that state even repeating the cycle with a new provider. The component with the 'X' had been installed and behaved like an OR rather than an AND for which the timer trigger operated the exception in any case. I can try to take it off and try it out.

Reply:
done